### DISCUSSION TITLE
Ranking Parameters

### DISCUSSION THREAD

**Discussion by MattiSG** [OpenTermsArchive/terms-types#16] [Added]
Several documents that are planned for addition in `p2b-compliance` are focused on algorithmic ranking and platform transparency and lack a dedicated document type. The name suggested by the team is “Ranking Parameters”. Examples given are:

- https://allegro.pl/pomoc/dla-sprzedajacych/wystawianie-oferty-przez-formularz/jestem-sprzedajacym-na-czym-polega-sortowanie-po-trafnosci-9dGRXKV4MTl
- https://sellercentral-europe.amazon.com/gp/help/external/GHEZQUQYZSPGXHG3?language=en_GB&ref=efph_GHEZQUQYZSPGXHG3_cont_G521
- https://support.microsoft.com/en-gb/topic/how-bing-delivers-search-results-d18fc815-ac37-4723-bc67-9229ce3eb6a3
- https://www.tripadvisor.com/pages/serviceEN.html
- https://studiosupport.trivago.com/hc/en-us/articles/4402598001809-Why-doesn-t-my-property-appear-in-trivago-search-results-
- https://help.yahoo.com/kb/search-for-desktop/higher-website-rank-sln2216.html

**Comment by MattiSG** [comment]
## Recap table of provided examples

| URL                                                                                                                                                                                                                    | Title                                                     | Service     | Describes rights and duties | Audience                                    |
|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------|-------------|-----------------------------|---------------------------------------------|
| [URL](https://sellercentral-europe.amazon.com/gp/help/external/GHEZQUQYZSPGXHG3?language=en_GB&ref=efph_GHEZQUQYZSPGXHG3_cont_G521)                                                                                           | How products are given visibility in the Amazon store     | Amazon      | No                          | Sellers                                     |
| [URL](https://support.microsoft.com/en-gb/topic/how-bing-delivers-search-results-d18fc815-ac37-4723-bc67-9229ce3eb6a3)                                                                                                        | How Bing delivers search results                          | Bing        | No                          | End users, website owners                   |
| [URL](https://www.tripadvisor.com/pages/serviceEN.html)                                                                                                                                                                       | How the site works                                        | TripAdvisor | Yes                         | Property & activity owners, reviewing users |
| [URL](https://studiosupport.trivago.com/hc/en-us/articles/4402598001809-Why-doesn-t-my-property-appear-in-trivago-search-results-)                                                                                            | Why doesn’t my property appear in trivago search results? | Trivago     | No                          | Property owners                             |
| [URL](https://help.yahoo.com/kb/search-for-desktop/higher-website-rank-sln2216.html)                                                                                                                                          | Get a higher website rank                                 | Yahoo       | No                          | Website owners                              |
| [URL](https://allegro-pl.translate.goog/pomoc/dla-sprzedajacych/wystawianie-oferty-przez-formularz/jestem-sprzedajacym-na-czym-polega-sortowanie-po-trafnosci-9dGRXKV4MTl?_x_tr_sl=pl&_x_tr_tl=en&_x_tr_hl=fr&_x_tr_pto=wapp) | I am the seller. What is relevancy sort?                  | Allegro     | Can be interpreted as yes                         | Sellers                                     |

## Relevance

With the exception of TripAdvisor and, to a lower extent, Allegro, these documents do not seem to be _contractual_ documents. They are explainers, or even performance tips.

While Open Terms Archive focuses on contractual documents, we already have some notable cases where we went beyond pure contractual commitment, for example with Community Guidelines, which were of major interest to our partners Reset and Platform Governance Archive and to the general public. I believe algorithmic ranking parameters fall in the same category.

## Tryptich

- Writer: `listing service provider`
- Audience: `users presented with a list of ranked results, or contributors of results that are ranked`
- Object: `methods and parameters for ranking results`

## Naming

The suggested “Ranking Parameters” is relevant. However, it describes the _contents_ of the document more than its _type_. Indeed, all the existing types include a noun such as “Guidelines”, “Conditions” or “Terms”. Given the examples, none of those would be appropriate since they are non-binding. On top, “Parameters”, while technically most accurate for algorithmic ranking, might not be very clear to all contributors, and does not include the description of the methods in which these parameters are taken into account.

I thus suggest, in my personal order of preference:

1. Ranking Method Description
2. Ranking Parameters Description
3. Ranking Method Disclosure
4. Ranking Parameters Disclosure

**Reply by martinratinaud** [reply]
Thanks for this deep analysis which I agree on.
I would go with number 1 also.

**Reply by lucielechardoy** [reply]
'Ranking parameters' is the wording coming from 2 pieces of legislation: 
- P2B Regulation 2019/1150, Article 5(1): 'Providers of online intermediation services shall set out in their terms and conditions **the main parameters determining ranking** and the reasons for the relative importance of those main parameters as opposed to other parameters.'
-'Omnibus Directive 2019/2161, Article 4(5): '1. Before a consumer is bound by a distance contract, or any corresponding offer, on an online marketplace, the provider of the online marketplace shall, without prejudice to Directive 2005/29/EC, provide the consumer with the following information in a clear and comprehensible manner and in a way appropriate to the means of distance communication: (a) general information, made available in a specific section of the online interface that is directly and easily accessible from the page where the offers are presented, on **the main parameters determining ranking**, as defined in point (m) of Article 2(1) of Directive 2005/29/EC, of offers presented to the consumer as a result of the search query and the relative importance of those parameters as opposed to other parameters' 

The P2B Regulation is about the information provided to business users of platforms, while the Omnibus Directive is about the information provided to consumers (end users) of platforms, but both require this information on ranking parameters

**Reply by MattiSG** [reply]
Thank you @lucielechardoy for these references! Considering that the legislation also requires the “relative importance” of those parameters and the reasons, I suggest the following:

- Type: `Ranking Parameters Description`
- Writer: `listing service provider`
- Audience: `users presented with a list of ranked results, or contributors of results that are ranked`
- Object: `parameters determining ranking and the reasons for their relative importance`
- References: `P2B Regulation 2019/1150, Article 5(1)`, `Omnibus Directive 2019/2161, Article 4(5)`, [point (m) of Article 2(1) of Directive 2005/29/EC](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX:02005L0029-20220528#tocId5)

**Reply by MattiSG** [reply]
Based on [Article 5 of P2B Regulation 2019/1150](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=celex%3A32019R1150#d1e868-57-1):

> 1.   Providers of online intermediation services shall set out in their terms and conditions the main parameters determining ranking and the reasons for the relative importance of those main parameters as opposed to other parameters.
> 2.   Providers of online search engines shall set out the main parameters, which individually or collectively are most significant in determining ranking and the relative importance of those main parameters, by providing an easily and publicly available description, drafted in plain and intelligible language, on the online search engines of those providers. They shall keep that description up to date.

I suggest to define `writer` more precisely as `search engine or intermediation service provider`.

**Reply by MattiSG** [reply]
Also, Article 4(5) of Omnibus Directive 2019/2161 is an amendment. I thus rewrite it to [Article 6a of Directive 2011/83/EU](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A02011L0083-20220528#tocId12).

**Comment by MattiSG** [comment]
This was added in https://github.com/OpenTermsArchive/engine/pull/927.
